### DISCUSSION TITLE
show keypad screen when alarm fired

### DISCUSSION THREAD

**Discussion by jmtuset** [Blackymas/NSPanel_HA_Blueprint#2032] [Ideas]
Hi
Is it possible to display the keypad screen when the alarm is pending or activated? This will allow you to enter the code quickly.
Another option I was thinking about is if there is a service that allows changing the current page, allowing an external automation to change the current page. This option is perhaps more generic and allows more options.

Best regards
Josep Maria

**Comment by edwardtfn** [comment]
Please let me know if this is what you are asking for: https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/howto.md#call-a-page-directly

**Reply by jmtuset** [reply]
Yes, this is it. I'll try it. Thank you so much!!

**Reply by cucpo** [reply]
Did you manage to call the keyboard directly with Disarm status? It doesn't work for me.

**Reply by jmtuset** [reply]
@cucpo 
Yes, I have done that when the alarm is armed and goes off (because a sensor is activated) in the nspanel I automatically show the keyboard to allow enter the code. I did it using nodered but it can be done with another system, you just need to call various services with the data.

In nodered I read the mqtt alarm/state topic. If it is activated or pending I call various services. In my case the nspanel is called **nspanel_1**

- call service **nspanel_1_entity_details_show** with this data ```` { "entity": "alarm_control_panel.alarma", "back_page": "home" } ````

That displays the alarm page. Then I launch the same commands that the nspanel does when pressing the disarm button. I have obtained these parameters by analyzing the source code.

- Call service **nspanel_1_commmand** witg data ````{ "cmd":"page keyb_num" }```` to show keyb_num page
- Call service **nspanel_1_commponent_val** data ````{"id":"Example text","val":"42"}````. Now I don't remember what 42 means
- Call service **nspanel_1_commponent_text** witg data ````{"id":"keyb_num.domain","txt":"alarm"}````. To setup the domain
- Call service **nspanel_1_commponent_text** witg data ````{"id":"keyb_num.key","txt":"disarm"}```` To setup the action to perform
- Call service **nspanel_1_commponent_text** witg data ````{"id":"keyb_num.value","txt":"click"}````  
- Call service **nspanel_1_commponent_text** witg data ````{"id":"keyb_num.title","txt":"Desarmar"}```` to setup the title of keypad

I wish this can help you

BR

**Comment by kavo91** [comment]
Is it possible to just have the keypad display the entire time easily? I am currently running two as dedicated keypads showing the Alarmo keypad card in a dedicated dashboard but it is extremely slow
